extension/storage/filestorage/extension.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestorage
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/extension/storage"
+)
+
+type localFileStorage struct {
+	directory string
+	timeout   time.Duration
+	logger    *zap.Logger
+	clients   []*fileStorageClient
+}
+
+// Ensure this storage extension implements the appropriate interface
+var _ storage.Extension = (*localFileStorage)(nil)
+
+func newLocalFileStorage(logger *zap.Logger, config *Config) (component.Extension, error) {
+	info, err := os.Stat(config.Directory)
+	if (err != nil && os.IsNotExist(err)) || !info.IsDir() {
+		return nil, fmt.Errorf("directory must exist: %v", err)
+	}
+
+	return &localFileStorage{
+		directory: filepath.Clean(config.Directory),
+		timeout:   config.Timeout,
+		logger:    logger,
+		clients:   []*fileStorageClient{},
+	}, nil
+}
+
+// Start does nothing
+func (lfs *localFileStorage) Start(context.Context, component.Host) error {
+	return nil
+}
+
+// Shutdown will close any open databases
+func (lfs *localFileStorage) Shutdown(context.Context) error {
+	for _, client := range lfs.clients {
+		client.close()
+	}
```

Comment:
Sounds like a chaos here, there is no clear owner of the components. Which component is responsible to close the "client" instance? Not well documented and based on the code seems a bit strange to say the least.

Reply:
Yeah, fair point. 

When I implemented this, I had in mind the concept of cleaning up stale storage files [#3149 (contrib)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3149), which I think would properly be the responsibility of this extension and might look similar to this implementation of tracking/closing clients. 

In any case, you are right - the clients can and should be handled by the components that requested them, and this expectation should be clearly documented.